packages/ticketing/src/pdf-tickets/googleWalletPass.ts

```diff
@@ -0,0 +1,399 @@
+  }
+
+  async upsertObject(classSuffix, objectSuffix, barcodeWithUrl) {
+    let response;
+    const newObject = {
+      id: `${this.issuerId}.${objectSuffix}`,
+      classId: `${this.issuerId}.${classSuffix}`,
+      state: 'ACTIVE',
+      ticketNumber: objectSuffix,
+      barcode: {
+        type: 'QR_CODE',
+        value: barcodeWithUrl,
+      },
+      passConstraints: {
+        screenshotEligibility: 'INELIGIBLE',
+      },
+    };
+    try {
+      response = await this.client.eventticketobject.get({
+        resourceId: `${this.issuerId}.${objectSuffix}`,
+      });
+    } catch (err) {
+      if (err.response && err.response.status === 404) {
+        console.log(`Object ${this.issuerId}.${objectSuffix} not found!`);
+        response = await this.client.eventticketobject.insert({
+          requestBody: newObject,
+        });
+        return `${this.issuerId}.${objectSuffix}`;
+      } else {
+        console.error(err);
+        return `${this.issuerId}.${objectSuffix}`;
+      }
+    }
+
+    const updatedObject = {
+      ...response.data,
+      ...newObject,
+    };
+
+    await this.client.eventticketobject.update({
+      resourceId: `${this.issuerId}.${objectSuffix}`,
+      requestBody: updatedObject,
+    });
+
+    return `${this.issuerId}.${objectSuffix}`;
+  }
+
+  async expireObject(objectSuffix) {
+    try {
+      await this.client.eventticketobject.get({
+        resourceId: `${this.issuerId}.${objectSuffix}`,
+      });
+    } catch (err) {
+      if (err.response && err.response.status === 404) {
+        console.log(`Object ${this.issuerId}.${objectSuffix} not found!`);
+        return `${this.issuerId}.${objectSuffix}`;
+      } else {
+        console.error(err);
+        return `${this.issuerId}.${objectSuffix}`;
+      }
+    }
+
+    const patchBody = {
+      state: 'EXPIRED',
+    };
+
+    await this.client.eventticketobject.patch({
+      resourceId: `${this.issuerId}.${objectSuffix}`,
+      requestBody: patchBody,
+    });
+
+    return `${this.issuerId}.${objectSuffix}`;
+  }
+
+  createJwtNewObjects(classSuffix, objectSuffix) {
+    const newClass = {
+      id: `${this.issuerId}.${classSuffix}`,
+    };
+    const newObject = {
+      id: `${this.issuerId}.${objectSuffix}`,
+      classId: `${this.issuerId}.${classSuffix}`,
+      state: 'ACTIVE',
+    };
+    const claims = {
+      iss: this.credentials.client_email,
+      aud: 'google',
+      origins: ['www.example.com'],
+      typ: 'savetowallet',
+      payload: {
+        eventTicketClasses: [newClass],
+        eventTicketObjects: [newObject],
+      },
+    };
+
+    const token = this.jwt.sign(claims, this.credentials.private_key, {
+      algorithm: 'RS256',
+    });
+
+    return `https://pay.google.com/gp/v/save/${token}`;
+  }
+}
+
+export default async (token, unchainedAPI: Context) => {
+  const GoogleWallet = new GoogleEventTicketWallet(process.env.GOOGLE_WALLET_ISSUER_ID as string);
```

Comment:
Import the deps at the top with top level await then test if deps are installed here, else throw here so that the server doesn't boot up when dev tries to use the google wallet renderer.